onnxruntime/core/providers/cuda/rnn/cudnn_rnn_base.cc

```diff
@@ -288,12 +317,15 @@ Status CudnnRnnBase<T>::ComputeInternal(OpKernelContext* ctx) const {
     }
   }
 
-  if (sequence_lens_data != nullptr && y_h_data != nullptr && y_data != nullptr) {
+  if ((CUDNN_RNN_RELU == rnn_mode_ || CUDNN_RNN_TANH == rnn_mode_) && sequence_lens_data != nullptr && y_h_data != nullptr && y_data != nullptr) {
+    auto count = sequence_lens->Shape().Size();
+    auto sequence_lens_gpu = GetScratchBuffer<int32_t>(count);
+    cudaMemcpyAsync(sequence_lens_gpu.get(), sequence_lens_data, count * sizeof(int32_t), cudaMemcpyHostToDevice);
```

Comment:
>cudaMemcpyAsync [](start = 4, length = 15)

seems sequence_lens_data may be stomped by CPU before consumed by GPU? I think you need to create a temp CPU copy for it and do the deferred release.

Another question is why this is not caught be test...